weed/server/filer_server_handlers.go

```diff
@@ -63,11 +63,11 @@ func (fs *FilerServer) filerHandler(w http.ResponseWriter, r *http.Request) {
 		stats.FilerRequestHistogram.WithLabelValues(stats.ChunkProxy).Observe(time.Since(start).Seconds())
 		return
 	}
-
-	defer func() {
-		stats.FilerRequestCounter.WithLabelValues(r.Method, strconv.Itoa(statusRecorder.Status)).Inc()
-		stats.FilerRequestHistogram.WithLabelValues(r.Method).Observe(time.Since(start).Seconds())
-	}()
+	requestMethod := r.Method
+	defer func(method *string) {
+		stats.FilerRequestCounter.WithLabelValues(*method, strconv.Itoa(statusRecorder.Status)).Inc()
+		stats.FilerRequestHistogram.WithLabelValues(*method).Observe(time.Since(start).Seconds())
+	}(&requestMethod)
```

Comment:
this would not work. The value for requestMethod is already passed in when `defer`.

Reply:
> this would not work. The value for requestMethod is already passed in when `defer`.

I double-checked it, it works, since a pointer to a variable is passed which can then be changed.